examples/solid/solid-start/src/routes/index.tsx

```diff
@@ -0,0 +1,63 @@
+import { Title } from 'solid-start'
+import { createQuery, useQueryClient } from '@tanstack/solid-query'
+import { createSignal, For, Suspense } from 'solid-js'
+import { isServer } from 'solid-js/web'
+
+interface PostData {
+  userId: number
+  id: number
+  title: string
+  body: string
+}
+
+export default function Home() {
+  const [postId, setPostId] = createSignal(1)
+
+  const query = createQuery(() => ({
+    queryKey: ['posts', postId()],
+    queryFn: async () => {
+      const response = await fetch(
+        'https://jsonplaceholder.typicode.com/posts/' + postId(),
+      ).then((res) => res.json())
+      return [response] as PostData[]
+    },
+  }))
+
+  return (
+    <main>
+      <Title>Solid Query v5</Title>
+      <h1>Solid Query v5</h1>
+      <p>
+        This is just a dev release for demo. Please dont use it in production
+        yet. Also this package will be deprecated as soon as features are live
+        on @tanstack/query
+      </p>
+      <div>
+        <button
+          onClick={() => {
+            setPostId((id) => (id === 1 ? 1 : id - 1))
+          }}
+        >
+          Previous Page
+        </button>
+        <button
+          onClick={() => {
+            setPostId((id) => (id === 100 ? 100 : id + 1))
+          }}
+        >
+          Next Page
+        </button>
+      </div>
+      <Suspense>
```

Comment:
I'm not sure that this `Suspense` boundary is accomplishing anything? Since the resource is created outside of the boundary, up on line 16, I believe that the suspense boundary that is catching the resource is actually the root one in `root.tsx`.

Can see this more clearly when you adjust `entry-server.tsx` to use `renderStream`, and turn off javascript to see which boundary actually catches the query on the server render:

<img width="451" alt="Screenshot 2023-01-13 at 1 41 41 PM" src="https://user-images.githubusercontent.com/847542/212395038-1486b745-e1ac-4f42-9118-2399a10c0b2f.png">

A better example might be to move the query into a separate posts component, so that one can see the streaming magic and suspense boundary at work, which is SO cool :):

<img width="1549" alt="Screenshot 2023-01-13 at 1 44 20 PM" src="https://user-images.githubusercontent.com/847542/212395552-d5f65bb7-1c8b-4648-9262-acfacd613c92.png">

And streaming shows off nicely - look at all that blue:

<img width="1728" alt="Screenshot 2023-01-13 at 1 47 16 PM" src="https://user-images.githubusercontent.com/847542/212396195-c0ec3dcc-7c52-4dfd-93e9-a879651b7a9d.png">

Example adjustment to this file:

```tsx
import { Title } from 'solid-start'
import { createQuery } from '@tanstack/solid-query'
import { createSignal, For, Suspense } from 'solid-js'
import type { Accessor } from 'solid-js'

interface PostData {
  userId: number
  id: number
  title: string
  body: string
}

export default function Home() {
  const [postId, setPostId] = createSignal(1)

  return (
    <main>
      <Title>Solid Query v5</Title>
      <h1>Solid Query v5</h1>
      <p>
        This is just a dev release for demo. Please dont use it in production
        yet. Also this package will be deprecated as soon as features are live
        on @tanstack/query
      </p>
      <div>
        <button
          onClick={() => {
            setPostId((id) => (id === 1 ? 1 : id - 1))
          }}
        >
          Previous Page
        </button>
        <button
          onClick={() => {
            setPostId((id) => (id === 100 ? 100 : id + 1))
          }}
        >
          Next Page
        </button>
      </div>

      <Suspense
        fallback={<div style={{ 'margin-top': '20px' }}>Loading post...</div>}
      >
        <Posts postId={postId} />
      </Suspense>
    </main>
  )
}

function Posts({ postId }: { postId: Accessor<number> }) {
  const query = createQuery(() => ({
    queryKey: ['posts', postId()],
    staleTime: Infinity,
    queryFn: async () => {
      const response = await fetch(
        'https://jsonplaceholder.typicode.com/posts/' + postId(),
      ).then((res) => res.json())

      // simulate extra latency to make things like streaming behavior more clear
      await new Promise((r) => setTimeout(r, 1000))

      return [response] as PostData[]
    },
  }))

  return (
    <For each={query.data}>
      {(post) => (
        <div>
          <h3>{post.title}</h3>
          <p>{post.body}</p>
        </div>
      )}
    </For>
  )
}
```

Reply:
Yeah this was from when i was trying to futz around with Streaming SSR but could not figure it out haha. I'll fix the example to follow the above changes you requested!